### ISSUE TITLE
Gradient for CUDA point sampling from rays

### ISSUE THREAD

**Issue by PNeigel** [sunset1995/DirectVoxGO#20]
Hi,

I've got a question regarding the CUDA extensions, more precise the following function:

https://github.com/sunset1995/DirectVoxGO/blob/9b172bfc9f0d349fb4b9db59caecd77b99ec7465/lib/cuda/render_utils_kernel.cu#L190

How difficult would it be to add a gradient for this function so that pytorch's autograd can backpropagate trough?

**Comment by sunset1995** [comment]
Let me try it.

**Comment by yashbhalgat** [comment]
Hi @sunset1995 @PNeigel, is there any update on this issue? I am trying to perform NeRF-- style pose refinement and I need to backpropagate through this function too

**Comment by wrld** [comment]
Is there any update? I met the same problem.
